src/backend/TrafficCourts/Workflow.Service/Sagas/VerifyEmailAddressSagaStateMachine.cs

```diff
@@ -100,11 +101,42 @@ public VerifyEmailAddressStateMachine(TimeProvider clock, ILogger<VerifyEmailAdd
         SetCompletedWhenFinalized();
     }
 
-    private void CreateToken(BehaviorContext<VerifyEmailAddressState, RequestEmailVerification> context)
+    private async void CreateTokenAndSendVerificationEmail(BehaviorContext<VerifyEmailAddressState, SubmitNoticeOfDispute> context)
     {
         var state = context.Saga;
+
+        // always save the ticket number
+        state.TicketNumber = context.Message.TicketNumber;
+
+        // cant send email address, if one is not supplied
+        if (string.IsNullOrEmpty(context.Message.EmailAddress))
+        {
+            LogNoEmailAddress(context);
+            return;
+        }
+
         state.EmailAddress = context.Message.EmailAddress;
+
+        bool optOutOfEmail = false; // TODO allow user opt out
+
+        if (optOutOfEmail)
+        {
+            LogOptOutOfEmail(context);
+            return;
+        }
```

Comment:
Should optOut logic be before checking if EmailAddress is null?

Reply:
I was not able to find "Opt Out" property on the Submit Notice of Dispute.

Reply:
Removed separate "opt out" logic